draft-ietf-ppm-dap.md

```diff
@@ -1291,44 +1295,70 @@ struct {
+in {{!HPKE, Section 7}}.
+
+Aggregators MUST allocate distinct `id` values for each `HpkeConfig` in an
 `HpkeConfigList`.
 
-The Client MUST abort if any of the following happen for any HPKE config
-request:
+The Client MUST abort if:
 
-* the GET request did not return a valid HPKE config list;
+* the response is not a valid HPKE config list;
 * the HPKE config list is empty; or
-* no HPKE config advertised by the Aggregator specifies a supported a KEM, KDF,
-  or AEAD algorithm triple.
-
-Aggregators SHOULD use HTTP caching to permit client-side caching of this
-resource {{!RFC5861}}. Aggregators SHOULD favor long cache lifetimes to avoid
-frequent cache revalidation, e.g., on the order of days. Aggregators can control
-this cached lifetime with the Cache-Control header, as in this example:
-
-~~~ http-message
-  Cache-Control: max-age=86400
+* no HPKE config advertised by the Aggregator specifies a supported KEM, KDF and
+  AEAD algorithm triple.
+
+Aggregators SHOULD use caching to permit client-side caching of this resource
+{{!RFC9111}}. Aggregators SHOULD favor long cache lifetimes to avoid frequent
+cache revalidation, e.g., on the order of days. Aggregators can control this
+cached lifetime with the Cache-Control header.
+
+Servers should choose a Cache-Control value appropriate to the lifetime of their
+keys. Clients SHOULD follow the usual caching semantics {{!RFC9111}} for HPKE
+configurations.
+
+Long cache lifetimes may result in Clients using stale HPKE configurations.
+Aggregators SHOULD continue to accept reports with old keys for at least twice
+the cache lifetime in order to avoid rejecting reports.
+
+#### Example
+
+~~~ http
+GET /leader/hpke_config
+Host: example.com
+
+HTTP/1.1 200
+Content-Type: application/dap-hpke-config-list
+Cache-Control: max-age=86400
+
+encoded([
+  struct {
+    id = 194,
+    kem_id = 0x0010,
+    kdf_id = 0x0001,
+    aead_id = 0x0001,
+    public_key = [0x01, 0x02, 0x03, 0x04, ...],
+  } HpkeConfig,
+  struct {
+    id = 17,
+    kem_id = 0x0020,
+    kdf_id = 0x0001,
+    aead_id = 0x0003,
+    public_key = [0x04, 0x03, 0x02, 0x01, ...],
+  } HpkeConfig,
+])
 ~~~
 
-Servers should choose a `max-age` value appropriate to the lifetime of their
-keys. Clients SHOULD follow the usual HTTP caching {{!RFC9111}} semantics for
-HPKE configurations.
-
-Note: Long cache lifetimes may result in Clients using stale HPKE
-configurations; Aggregators SHOULD continue to accept reports with old keys for
-at least twice the cache lifetime in order to avoid rejecting reports.
-
 ### Upload Request
 
-Clients upload reports by using an HTTP POST to
-`{leader}/tasks/{task-id}/reports`. The payload is a `Report`, with media type
-"application/dap-report", structured as follows:
+Clients upload reports by sending a POST to `{leader}/tasks/{task-id}/reports`.
+The body is a `Report`, with media type "application/dap-report", structured as
+follows:
```

Comment:
no need to remind readers that POST is an HTTP request, and "body" is more appropriate than "payload"